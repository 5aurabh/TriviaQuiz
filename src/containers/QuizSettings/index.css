.section {
  margin-top: 50px;
}

.sectionHeading {
  font-size: 1.1rem;
}

.optionSection {
  font-size: 1rem;
  margin: 10px 0px;
}

.submitSection {
  margin-top: 60px;
  text-align: center;
}
