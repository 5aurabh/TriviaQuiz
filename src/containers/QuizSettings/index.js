// @flow
import React, { Component } from 'react';
import { Grid, Row, Col } from 'react-flexbox-grid';
import { SquareRadioButton, Button } from '../../components';
import * as constants from '../../commons/constants';
import styles from './index.css';
type difficulty = constants.EASY | constants.MEDIUM | constants.HARD;
type duration = constants.SHORT | constants.RESPECTABLE | constants.LENGTHY;

type Props = {
  startQuiz: () => *,
};

type State = {
  difficulty: ?difficulty,
  duration: ?duration,
};
class QuizSettings extends Component<Props, State> {
  state = {
    difficulty: null,
    duration: null,
  };
  handleDifficultyChange = (e: Event) => {
    const difficulty = e.currentTarget.value;
    this.setState({ difficulty });
  };
  handleDurationChange = (e: Event) => {
    const duration = e.currentTarget.value;
    this.setState({ duration });
  };

  startQuiz = () => {
    this.props.startQuiz(this.state.duration, this.state.difficulty);
  };
  render() {
    return (
      <div>
        <h3>
          <center>Welcome!</center>
        </h3>
        <div className={styles.section}>
          <div className={styles.sectionHeading}>
            How competitive you are feeling right now?
          </div>
          <div className={styles.optionSection}>
            <Grid fluid>
              <Row>
                <Col xs={12} md={4}>
                  <SquareRadioButton
                    name="difficulty"
                    id="diff-1"
                    label="Be easy on me"
                    value={constants.EASY}
                    onChange={this.handleDifficultyChange}
                  />
                </Col>
                <Col xs={12} md={4}>
                  <SquareRadioButton
                    name="difficulty"
                    id="diff-2"
                    label="I may win"
                    value={constants.MEDIUM}
                    onChange={this.handleDifficultyChange}
                  />
                </Col>
                <Col xs={12} md={4}>
                  <SquareRadioButton
                    name="difficulty"
                    id="diff-3"
                    label="Ask me anything"
                    value={constants.HARD}
                    onChange={this.handleDifficultyChange}
                  />
                </Col>
              </Row>
            </Grid>
          </div>
        </div>
        <div className={styles.section}>
          <div className={styles.sectionHeading}>
            How much time you can spare?
          </div>
          <div className={styles.optionSection}>
            <Grid fluid>
              <Row>
                <Col xs={12} md={4}>
                  <SquareRadioButton
                    name="time"
                    id="time-1"
                    label="Be quick"
                    value={constants.SHORT}
                    onChange={this.handleDurationChange}
                  />
                </Col>
                <Col xs={12} md={4}>
                  <SquareRadioButton
                    name="time"
                    id="time-2"
                    label="Till I finish my coffee"
                    value={constants.RESPECTABLE}
                    onChange={this.handleDurationChange}
                  />
                </Col>
                <Col xs={12} md={4}>
                  <SquareRadioButton
                    name="time"
                    id="time-3"
                    label="I am all here for you"
                    value={constants.LENGTHY}
                    onChange={this.handleDurationChange}
                  />
                </Col>
              </Row>
            </Grid>
          </div>
          <div className={styles.submitSection}>
            <Button
              text="Start"
              backgroundColor="#29CF66"
              onClick={this.startQuiz}
              disabled={
                this.state.duration && this.state.difficulty ? false : true
              }
            />
          </div>
        </div>
      </div>
    );
  }
}
export default QuizSettings;
