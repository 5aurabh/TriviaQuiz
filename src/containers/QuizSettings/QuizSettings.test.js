import React from 'react';
import renderer from 'react-test-renderer';
import QuizSettings from './index';

it('Quizsettings: default', () => {
  const component = renderer.create(<QuizSettings />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
