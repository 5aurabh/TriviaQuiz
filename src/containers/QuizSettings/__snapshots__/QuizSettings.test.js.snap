// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quizsettings: default 1`] = `
<div>
  <h3>
    <center>
      Welcome!
    </center>
  </h3>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      How competitive you are feeling right now?
    </div>
    <div
      className={undefined}
    >
      <div
        className="container-fluid"
      >
        <div
          className="row"
        >
          <div
            className="col-xs-12 col-md-4"
          >
            <div
              className={undefined}
            >
              <input
                className={undefined}
                defaultChecked={false}
                id="diff-1"
                name="difficulty"
                onChange={[Function]}
                type="radio"
                value="EASY"
              />
              <label
                htmlFor="diff-1"
              >
                Be easy on me
              </label>
            </div>
          </div>
          <div
            className="col-xs-12 col-md-4"
          >
            <div
              className={undefined}
            >
              <input
                className={undefined}
                defaultChecked={false}
                id="diff-2"
                name="difficulty"
                onChange={[Function]}
                type="radio"
                value="MEDIUM"
              />
              <label
                htmlFor="diff-2"
              >
                I may win
              </label>
            </div>
          </div>
          <div
            className="col-xs-12 col-md-4"
          >
            <div
              className={undefined}
            >
              <input
                className={undefined}
                defaultChecked={false}
                id="diff-3"
                name="difficulty"
                onChange={[Function]}
                type="radio"
                value="HARD"
              />
              <label
                htmlFor="diff-3"
              >
                Ask me anything
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      How much time you can spare?
    </div>
    <div
      className={undefined}
    >
      <div
        className="container-fluid"
      >
        <div
          className="row"
        >
          <div
            className="col-xs-12 col-md-4"
          >
            <div
              className={undefined}
            >
              <input
                className={undefined}
                defaultChecked={false}
                id="time-1"
                name="time"
                onChange={[Function]}
                type="radio"
                value="SHORT"
              />
              <label
                htmlFor="time-1"
              >
                Be quick
              </label>
            </div>
          </div>
          <div
            className="col-xs-12 col-md-4"
          >
            <div
              className={undefined}
            >
              <input
                className={undefined}
                defaultChecked={false}
                id="time-2"
                name="time"
                onChange={[Function]}
                type="radio"
                value="RESPECTABLE"
              />
              <label
                htmlFor="time-2"
              >
                Till I finish my coffee
              </label>
            </div>
          </div>
          <div
            className="col-xs-12 col-md-4"
          >
            <div
              className={undefined}
            >
              <input
                className={undefined}
                defaultChecked={false}
                id="time-3"
                name="time"
                onChange={[Function]}
                type="radio"
                value="LENGTHY"
              />
              <label
                htmlFor="time-3"
              >
                I am all here for you
              </label>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className={undefined}
    >
      <button
        className="undefined "
        disabled={true}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "grey",
            "color": "#fff",
            "cursor": "not-allowed",
          }
        }
      >
        Start
      </button>
    </div>
  </div>
</div>
`;
